Sources/PluginLibrary/Descriptor.swift

```diff
@@ -388,7 +388,17 @@ public final class FieldDescriptor {
   /// When this is a enum field, the enum's desciptor.
   public private(set) weak var enumType: EnumDescriptor!
 
-  public var isPacked: Bool { return proto.options.packed }
+  /// Should this field be packed format.
+  public var isPacked: Bool {
+    // This logic comes from the C++ FieldDescriptor::is_packed() impl.
+    guard isPackable else { return false }
+    if file.syntax == .proto2 {
+      return proto.hasOptions && proto.options.packed
+    } else {
+      return !proto.hasOptions || !proto.options.hasPacked || proto.options.packed
```

Comment:
It might be worth a comment here that calls out the fact that packed is the default for proto3 with a link to <https://developers.google.com/protocol-buffers/docs/proto3#specifying-field-rules> to save future generations a "🤔"